### PR TITLE
Clever OAuth sign-up/sign-in integration tests

### PR DESCRIPTION
**Pull request by islemaster**
Consider this a follow-on to https://github.com/code-dot-org/code-dot-org/pull/25062.  Adds matching tests for Clever, and does significant cleanup on the Google tests as well, for readability.

New test cases:
- Clever student sign-up
- Clever teacher sign-up
- Clever student sign-up (new flow)
- Clever teacher sign-up (new flow)
- Clever student sign-in
- Clever teacher sign-in

### CODE REVIEW

**Comment by islemaster** [comment]
Any thoughts on where to put a helpers file to share helper methods between these two tests?

**Comment by islemaster** [comment]
Looking through coverage reports from the Clever integration tests and these do a pretty good job of hitting all possible branches.  Things I've noticed are missed:

We're missing the second half of RegistrationsController#new, but we've already discussed this bug:
image


The GDPR data transfer agreement field isn't covered.
image

The Clever email-already-taken path, and the (maybe unreachable in practice?) old finish registration path where the details we get from Clever aren't sufficient to create the user:
image

Any and all takeover code (Clever teacher forced takeover, Google silent takeover, etc).

Clever admin sign-ups:
image

File: dashboard/test/controllers/omniauth_callbacks_controller/google_oauth2_test.rb [hunk 1]
```diff
@@ -255,6 +163,19 @@ def generate_auth_hash(args = {})
       )
     end
 
+    # The user signs in through Google
+    # The oauth endpoint (which is mocked) redirects to the oauth callback,
+    # which in turn does some work and redirects to something else: homepage, finish_sign_up, etc.
+    def sign_in_through_google
```
Comment (maddiedierker) [inline dashboard/test/controllers/omniauth_callbacks_controller/google_oauth2_test.rb:169]: love this cleanup!

**Comment by maddiedierker** [comment]
re: where to put a helper file for these tests -- i think either `dashboard/test/helpers/` or `dashboard/test/controllers/omniauth_callbacks_controller/` would work (slight preference for the former)

**Comment by islemaster** [comment]
Hmm... `dashboard/test/helpers` seems to contain tests for `dashboard/app/helpers`, not test-helper code.  Maybe I'll ask the group.

**Pull request merged**